File: common/src/main/java/dev/number6/slack/dagger/SlackClientModule.java
package dev.number6.slack.dagger;

import dev.number6.slack.adaptor.SlackClientAdaptor;
import dev.number6.slack.port.HttpPort;
import dev.number6.slack.port.SlackPort;
import dagger.Module;
import dagger.Provides;

@Module
public class SlackClientModule {

    @Provides
    public SlackPort providesSlackPort(HttpPort client) {
        return new SlackClientAdaptor(client);
    }
}

File: common/src/main/java/dev/number6/slack/dagger/SlackClientModule.kt
package dev.number6.slack.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.adaptor.SlackClientAdaptor
import dev.number6.slack.port.HttpPort
import dev.number6.slack.port.SlackPort

@Module
class SlackClientModule {
    @Provides
    fun providesSlackPort(client: HttpPort): SlackPort {
        return SlackClientAdaptor(client)
    }
}